source/plugins/ruby/in_kube_pvinventory.rb

```diff
@@ -0,0 +1,253 @@
+module Fluent
+  class Kube_PVInventory_Input < Input
+    Plugin.register_input("kubepvinventory", self)
+
+    @@hostName = (OMS::Common.get_hostname)
+
+    def initialize
+      super
+      require "yaml"
+      require "yajl/json_gem"
+      require "yajl"
+      require "time"
+      require_relative "KubernetesApiClient"
+      require_relative "ApplicationInsightsUtility"
+      require_relative "oms_common"
+      require_relative "omslog"
+      require_relative "constants"
+
+      # Response size is around 1500 bytes per PV
+      @PV_CHUNK_SIZE = "5000"
```

Comment:
How much (in terms of bytes) does 5k PVs attribute to in the payload ?

Reply:
From testing, there's an average response size of about 1500 bytes per PV in the payload. So 7.5 MB for 5k PVs. I kept the chunk size very conservative since I wasn't sure if customers might have longer names than I did for the PV, PVC, disk, or share file. We can increase it though.

Reply:
look fine then. thanks.